sklearn/preprocessing/tests/test_data.py

```diff
@@ -119,6 +120,27 @@ def test_polynomial_features():
     assert_array_almost_equal(X_poly, P2[:, [0, 1, 2, 4]])
 
 
+def test_polynomial_feature_names():
+    X = np.arange(30).reshape(10, 3)
+    poly = PolynomialFeatures(degree=2, include_bias=True).fit(X)
+    feature_names = poly.get_feature_names()
+    assert_array_equal(['1', 'x0^1', 'x1^1', 'x2^1', 'x0^2', 'x0^1 x1^1',
+                        'x0^1 x2^1', 'x1^2', 'x1^1 x2^1', 'x2^2'],
+                       feature_names)
+
+    poly = PolynomialFeatures(degree=3, include_bias=False).fit(X)
+    feature_names = poly.get_feature_names(["a", "b", "c"])
+    assert_array_equal(['a^1', 'b^1', 'c^1', 'a^2', 'a^1 b^1', 'a^1 c^1', 'b^2',
+                        'b^1 c^1', 'c^2', 'a^3', 'a^2 b^1', 'a^2 c^1',
+                        'a^1 b^2', 'a^1 b^1 c^1', 'a^1 c^2', 'b^3', 'b^2 c^1',
+                        'b^1 c^2', 'c^3'], feature_names)
+    # test some unicode
+    poly = PolynomialFeatures(degree=1, include_bias=True).fit(X)
+    feature_names = poly.get_feature_names([u"\u0001F40D", u"\u262E", u"\u05D0"])
+    assert_array_equal(["1", u"\u0001F40D^1", u"\u262E^1", u"\u05D0^1"],
```

Comment:
It's a bit awkward that 1 is not `unicode`. Can we find a simple convention for such things?